web_src/js/components/RepoContributors.vue

```diff
@@ -0,0 +1,353 @@
+        a[sortingCriteria] > b[sortingCriteria] ? -1 : a[sortingCriteria] === b[sortingCriteria] ? 0 : 1
+      ).slice(0, 100);
+    },
+    async fetchGraphData() {
+      this.isLoading = true;
+      fetch(`${this.repoLink}/activity/contributors/data`)
+        .then((response) => response.json())
+        .then((data) => {
+          const {total, ...rest} = data;
+          this.contributorsStats = rest;
+          this.dateFrom = new Date(total.weeks[0].week);
+          this.dateUntil = new Date();
+          this.startDate = this.dateFrom;
+          this.endDate = this.dateUntil;
+          this.sortContributors();
+          this.totalStats = total;
+          this.errorText = '';
+        })
+        .catch((e) => {
+          this.errorText = e.message;
+        })
+        .finally(() => {
+          this.isLoading = false;
+        });
+    },
+
+    filterContributorWeeksByDateRange() {
+      const filteredData = {};
+
+      const data = this.contributorsStats;
+      for (const key of Object.keys(data)) {
+        const user = data[key];
+        user['total_commits'] = 0;
+        user['total_additions'] = 0;
+        user['total_deletions'] = 0;
+        user['max_contribution_type'] = 0;
+        const filteredWeeks = user.weeks.filter((week) => {
+          const weekDate = new Date(week.week);
+          if (weekDate >= this.dateFrom && weekDate <= this.dateUntil) {
+            user['total_commits'] += week.commits;
+            user['total_additions'] += week.additions;
+            user['total_deletions'] += week.deletions;
+            if (week[this.type] > user['max_contribution_type']) {
+              user['max_contribution_type'] = week[this.type];
+            }
+            return true;
+          } return false;
+        });
+
+        filteredData[key] = {...user, weeks: filteredWeeks};
+      }
+
+      return filteredData;
+    },
+
+    maxMainGraph() {
+      const maxValue = Math.max(
+        ...this.totalStats.weeks.map((o) => o[this.type])
+      );
+      const [cooefficient, exp] = maxValue
+        .toExponential()
+        .split('e')
+        .map(Number);
+      if (cooefficient % 1 === 0) {
+        return maxValue;
+      }
+      return (1 - (cooefficient % 1)) * 10 ** exp + maxValue;
+    },
+    maxContributorGraph() {
+      const maxValue = Math.max(
+        ...this.sortedContributors.map((c) => c['max_contribution_type'])
+      );
+      const [cooefficient, exp] = maxValue
+        .toExponential()
+        .split('e')
+        .map(Number);
+      if (cooefficient % 1 === 0) {
+        return maxValue;
+      }
+      return (1 - (cooefficient % 1)) * 10 ** exp + maxValue;
+    },
+
+    toGraphData(data) {
+      const style = getComputedStyle(document.body);
+      const colorName = this.type === 'commits' ? '--color-primary-alpha-60' : (this.type === 'additions' ? '--color-green-badge-hover-bg' : '--color-red-badge-hover-bg');
+      const color = style.getPropertyValue(colorName);
```

Comment:
```suggestion
      const color = style.getPropertyValue(colorName).trim();
```

IIRC, this may be needed in some cases.